contracts/contracts/strategies/plume/RoosterAMOStrategy.sol

```diff
@@ -0,0 +1,1274 @@
+            _totalVaultValue.divPrecisely(_totalOethpSupply) <
+            SOLVENCY_THRESHOLD
+        ) {
+            revert("Protocol insolvent");
+        }
+    }
+
+    /**
+     * @dev Collect stkAave, convert it to AAVE send to Vault.
+     */
+    function collectRewardTokens()
+        external
+        override
+        onlyHarvester
+        nonReentrant
+    {
+        // Do nothing if there's no position minted
+        if (tokenId > 0) {
+            uint32[] memory binIds = new uint32[](1);
+            binIds[0] = 1;
+
+            uint256 lastEpoch = votingDistributor.lastEpoch();
+
+            poolDistributor.claimLp(
+                address(this),
+                tokenId,
+                mPool,
+                binIds,
+                lastEpoch
+            );
+        }
+
+        // Run the internal inherited function
+        _collectRewardTokens();
+    }
+
+    /***************************************
+            Balances and Fees
+    ****************************************/
+
+    /**
+     * @dev Get the total asset value held in the platform
+     * @param _asset      Address of the asset
+     * @return balance    Total value of the asset in the platform
+     */
+    function checkBalance(address _asset)
+        external
+        view
+        override
+        returns (uint256)
+    {
+        require(_asset == WETH, "Only WETH supported");
+
+        // because of PoolLens inaccuracy there is usually some dust WETH left on the contract
+        uint256 _wethBalance = IERC20(WETH).balanceOf(address(this));
+        // just paranoia check, in case there is OETHb in the strategy that for some reason hasn't
+        // been burned yet. This should always be 0.
+        uint256 _oethpBalance = IERC20(OETHp).balanceOf(address(this));
+        return underlyingAssets + _wethBalance + _oethpBalance;
+    }
+
+    /**
+     * @notice Strategy reserves (which consist only of WETH in case of Rooster - Plume pool)
+     * when the tick price is closest to parity - assuring the lowest amount of tokens
+     * returned for the current position liquidity.
+     */
+    function _balanceInPosition() internal view returns (uint256 _wethBalance) {
+        (, , uint256 liquidity) = _getPositionInformation();
+
+        uint256 _oethbBalance;
+
+        (
+            _wethBalance,
+            _oethbBalance
+        ) = _reservesInTickForGivenPriceAndLiquidity(
+            sqrtPriceTickLower,
+            sqrtPriceTickHigher,
+            sqrtPriceAtParity,
+            liquidity
+        );
+
+        require(_oethbBalance == 0, "Non zero oethbBalance");
+    }
+
+    /**
+     * @notice Tick dominance denominated in 1e18
+     *
+     */
+    function tickDominance() public view returns (uint256 _tickDominance) {
+        uint256 _currentPrice = getPoolSqrtPrice();
+        (
+            IMaverickV2Pool.TickState memory tickState,
+            ,
+
+        ) = _reservesInTickForGivenPrice(tickNumber, _currentPrice);
```

Comment:
This can be replaced by something like this as we only used the tickState, and `_currentPrice` is not used: 
```solidity
IMaverickV2Pool.TickState memory tickState = mPool.getTick(tickNumber).
```